kirin/piv/model_maker.py

```diff
@@ -193,13 +225,23 @@ def build_trip_updates(self, rt_update):
         if not events and not plan_transport_source:
             raise InvalidArguments('No object "evenement" or "planTransportSource" available in feed provided')
 
+        higher_event = ujson.loads('{"type": "UNDEFINED", "texte": ""}')
         if events:
-            event_types = [get_value(event, "type", nullable=True) for event in events]
-            if "RETARD" not in event_types:
-                raise UnauthorizedValue("None of the event-types {} are supported".format(event_types))
+            for event in events:
+                event_type = get_value(event, "type", nullable=True)
+                if event_type and event_type in trip_piv_status_to_effect and event_type in MANAGED_EVENTS:
+                    higher_event = (
+                        event
+                        if _get_trip_effect_order_from_piv_status(event_type)
+                        < _get_trip_effect_order_from_piv_status(get_value(higher_event, "type"))
+                        else higher_event
+                    )
+            if trip_piv_status_to_effect[higher_event.get("type")] == TripEffect.UNDEFINED:
+                raise UnsupportedValue("None of the event-types {} are supported".format(events))
         elif plan_transport_source and plan_transport_source not in ["PTP", "OPE"]:
-            raise UnauthorizedValue("planTransportSource {} is not supported".format(plan_transport_source))
+            raise UnsupportedValue("planTransportSource {} is not supported".format(plan_transport_source))
 
+        json_train["evenement"] = higher_event
```

Comment:
To keep the same struct, as it's a list initially :
```suggestion
        json_train["evenement"] = [higher_event]
```
And a `[0]` would be required when using it after.